### PR TITLE
Fix rewards

### PR DESCRIPTION
**Pull request by nz-neooo**

### CODE REVIEW

File: contracts/RewardsDistributer.sol [hunk 1]
```diff
@@ -142,6 +142,32 @@ contract RewardsDistributer is IRewardsDistributer, Initializable, OwnableUpgrad
...
+     * @notice Reset era reward remove table.
+     * Only Owner can call.
+     * @param indexers address[]
+     * @param eras uint256[]
+     * @param addAmounts uint256[]
+     * @param removeAmounts uint256[]
+     */
+    function resetEraRewardRemoveTables(
+        address[] memory indexers,
+        uint256[] memory eras,
```
Comment (mzxyz) [inline contracts/RewardsDistributer.sol:154]: Shoud `uint256[] memory eras,` be era number other than era array?

Comment (nz-neooo) [reply contracts/RewardsDistributer.sol:154]: use `array` to batch many operations in different eras

**Comment by mzxyz** [comment]
- [x] Remove `addTable` in new method
- [x] Add new method to get rewardDebt

File: contracts/RewardsDistributer.sol [hunk 2]
```diff
@@ -142,6 +142,28 @@ contract RewardsDistributer is IRewardsDistributer, Initializable, OwnableUpgrad
...
+        uint256[] memory removeAmounts
+    ) external onlyOwner {
+        require(indexers.length == eras.length, 'G100');
+        require(indexers.length == removeAmounts.length, 'G100');
+
+        for (uint256 i = 0; i < indexers.length; i++) {
+            info[indexers[i]].eraRewardRemoveTable[eras[i]] = removeAmounts[i];
+
+            emit RewardsChanged(indexers[i], eras[i], 0, removeAmounts[i]);
```
Comment (ianhe8x) [inline contracts/RewardsDistributer.sol:163]:
```
if (removeAmounts[i] > nfo[indexers[i]].eraRewardRemoveTable[eras[i]]) {
  emit RewardsChanged(indexers[i], eras[i], 0, removeAmounts[i]-info[indexers[i]].eraRewardRemoveTable[eras[i]]);
} else {
  emit RewardsChanged(indexers[i], eras[i], info[indexers[i]].eraRewardRemoveTable[eras[i]]-removeAmounts[i], 0);
}
```

**Pull request merged**